pkg/controller/ipam/metrics.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipam
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+
+	crdv1a2 "antrea.io/antrea/pkg/apis/crd/v1alpha2"
+	"antrea.io/antrea/pkg/client/clientset/versioned"
+	crdinformers "antrea.io/antrea/pkg/client/informers/externalversions"
+	"antrea.io/antrea/pkg/client/informers/externalversions/crd/v1alpha2"
+	crdlisters "antrea.io/antrea/pkg/client/listers/crd/v1alpha2"
+	"antrea.io/antrea/pkg/ipam/poolallocator"
+)
+
+const (
+	controllerName        = "AntreaIPAMController"
+	metricNamespaceAntrea = "antrea"
+	metricSubsystemIPAM   = "ipam"
+)
+
+type AntreaIPAMMetricsHandler struct {
+	crdClient      versioned.Interface
+	ipPoolLister   crdlisters.IPPoolLister
+	ipPoolInformer v1alpha2.IPPoolInformer
+}
+
+var (
+	ipPoolUsageTotal = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Namespace:      metricNamespaceAntrea,
+			Subsystem:      metricSubsystemIPAM,
+			Name:           "ippool_usage_total",
+			Help:           "Total number of available IP addresses for the address pool.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"pool_name"},
+	)
+	ipPoolUsageUsed = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Namespace:      metricNamespaceAntrea,
+			Subsystem:      metricSubsystemIPAM,
+			Name:           "ippool_usage_used",
+			Help:           "Number of allocated IP addresses for the address pool.",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"pool_name"},
+	)
+)
+
+func NewAntreaIPAMMetricsHandler(crdClient versioned.Interface, crdInformerFactory crdinformers.SharedInformerFactory) *AntreaIPAMMetricsHandler {
+	legacyregistry.MustRegister(ipPoolUsageTotal)
+	legacyregistry.MustRegister(ipPoolUsageUsed)
+
+	ipPoolInformer := crdInformerFactory.Crd().V1alpha2().IPPools()
+	mh := &AntreaIPAMMetricsHandler{
+		crdClient:      crdClient,
+		ipPoolInformer: ipPoolInformer,
+		ipPoolLister:   ipPoolInformer.Lister()}
+	return mh
+}
+
+func (c *AntreaIPAMMetricsHandler) Run(stopCh <-chan struct{}) {
+	c.ipPoolInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
```

Comment:
nit: I notice many places in the code use `AddEventHandlerWithResyncPeriod` while supplying 0 for `resyncPeriod`, but I think you can just use `AddEventHandler` without the `resyncPeriod` parameter